pkg/kubelet/rkt/rkt.go

```diff
@@ -425,6 +436,77 @@ func (r *Runtime) makePodManifest(pod *api.Pod, pullSecrets []api.Secret) (*appc
 	var globalPortMappings []kubecontainer.PortMapping
 	manifest := appcschema.BlankPodManifest()
 
+	listReq := &rktapi.ListPodsRequest{
+		Filter: &rktapi.PodFilter{
+			States: []rktapi.PodState{
+				//TODO: In the future some pods can remain running after some apps exit: https://github.com/appc/spec/pull/500
+				rktapi.PodState_POD_STATE_RUNNING,
+				rktapi.PodState_POD_STATE_EXITED,
+				rktapi.PodState_POD_STATE_DELETING,
+				rktapi.PodState_POD_STATE_GARBAGE,
+			},
+			Annotations: []*rktapi.KeyValue{
+				&rktapi.KeyValue{
+					Key:   k8sRktKubeletAnno,
+					Value: k8sRktKubeletAnnoValue,
+				},
+				&rktapi.KeyValue{
+					Key:   k8sRktUIDAnno,
+					Value: string(pod.UID),
+				},
+			},
+		},
+	}
+	listResp, err := r.apisvc.ListPods(context.Background(), listReq)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't list pods: %v", err)
+	}
+
+	restartCount := 0
+	for _, rktpod := range listResp.Pods {
+		manifest := appcschema.PodManifest{}
+		err = json.Unmarshal(rktpod.Manifest, manifest)
```

Comment:
In the meantime would the correct thing to do be send a `InspectPodRequest` for each pod here?